src/app/WriteClient.cpp

```diff
@@ -176,26 +170,164 @@ TLV::TLVWriter * WriteClient::GetAttributeDataIBTLVWriter()
     return mWriteRequestBuilder.GetWriteRequests().GetAttributeDataIBBuilder().GetWriter();
 }
 
-CHIP_ERROR WriteClient::FinalizeMessage(System::PacketBufferHandle & aPacket)
+CHIP_ERROR WriteClient::FinalizeMessage(bool aHasMoreChunks)
 {
+    System::PacketBufferHandle packet;
     VerifyOrReturnError(mState == State::AddAttribute, CHIP_ERROR_INCORRECT_STATE);
+
+    TLV::TLVWriter * writer = mWriteRequestBuilder.GetWriter();
+    ReturnErrorCodeIf(writer == nullptr, CHIP_ERROR_INCORRECT_STATE);
+    ReturnErrorOnFailure(writer->UnreserveBuffer(kReservedSizeForTLVEncodingOverhead));
+
     AttributeDataIBs::Builder & attributeDataIBsBuilder = mWriteRequestBuilder.GetWriteRequests().EndOfAttributeDataIBs();
     ReturnErrorOnFailure(attributeDataIBsBuilder.GetError());
 
-    mWriteRequestBuilder.IsFabricFiltered(false).EndOfWriteRequestMessage();
```

Comment:
Still need a followup to remove the IsFabricFiltered thing from WriteRequestBuilder. It shouldn't be there at all.